### DISCUSSION TITLE
Simultaneously play existing midi file and real time input

### DISCUSSION THREAD

**Discussion by pianonyy** [mido/mido#437] [Q&A] [answered]
Hello!
I’m trying to realize real time python module which can play input midi events, and play existing midi files simultaneously. So in this way we can play something in real-time for existing midi patterns. But, I face the problem of outport it can’t handle several midi event sequences, or I ‘m doing something wrong. 

Please, could you help me to understand, how to organise multiple midi outputs using mido simultaneously?

**Comment by r1yk** [comment, marked as answer]
Here's an example where a MIDI file and real-time input are both sent to the same MIDI output. This code will run until the specified MIDI file has finished playing.
```python
import mido

output_device = mido.open_output('my_output_device')
realtime_input = mido.open_input('my_input_device')

# Messages created by realtime_input are sent to the output as they are received
realtime_input.callback = output_device.send

# Messages yielded by MidiFile.play are ALSO sent to the output
for message in mido.MidiFile('/path/to/midi_file.mid').play():
    output_device.send(message)
```
This works if you have only one output device, but there can be some weird artifacts from doing it this way. Your real-time input and the MIDI file could send conflicting messages to the output and cancel each other out. For example, if you're sustaining a note on the real-time input and the MIDI file sends a `note_off` message on the same note, your sustained note is also going to be turned off. If you have multiple MIDI devices available, you can avoid that by opening multiple outputs with mido. One handles the input from the real-time controller, and one handles the MIDI file. Here's the same example with two output devices:
```python
import mido

realtime_output = mido.open_output('my_realtime_output_device')
realtime_input = mido.open_input('my_input_device')

# Messages created by realtime_input are sent to the realtime_output as they are received
realtime_input.callback = realtime_output.send

# Create another output for handling a MIDI file
midi_file_output = mido.open_output('my_midi_file_output_device')
for message in mido.MidiFile('/path/to/midi_file.mid').play():
    midi_file_output.send(message)
```
